rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -363,16 +363,12 @@ bool Player::play_next()
   rosbag2_storage::SerializedBagMessageSharedPtr message_ptr = peek_next_message_from_queue();
 
   bool next_message_published = false;
-  while (message_ptr != nullptr && !next_message_published) {
-    if (play_until_timestamp_ >= starting_time_ &&
-      message_ptr->time_stamp > play_until_timestamp_)
-    {
-      break;
-    }
-    {
-      next_message_published = publish_message(message_ptr);
-      clock_->jump(message_ptr->time_stamp);
-    }
+  while (rclcpp::ok() && !next_message_published &&
+    message_ptr != nullptr && !shall_stop_at_timestamp(message_ptr->time_stamp))
```

Comment:
Not sure if you considered unifying the condition in a function. This condition is pretty similar to the one in play_messages_from_queue(). Perhaps it makes sense to encapsulate it in a function because you can benefit from passing message_ptr and evaluating the timestamp conditions as well.